app/backend/src/couchers/servicers/account.py

```diff
@@ -150,3 +157,61 @@ def MarkContributorFormFilled(self, request, context):
             user = session.query(User).filter(User.id == context.user_id).one()
             user.filled_contributor_form = request.filled_contributor_form
         return empty_pb2.Empty()
+
+    def ChangePhone(self, request, context):
+        phone = request.phone
+        # early quick validation
+        if phone and not is_e164_format(phone):
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.INVALID_PHONE)
+
+        with session_scope() as session:
+            user = session.query(User).filter(User.id == context.user_id).one()
+            if not phone:
+                user.phone = None
+                user.phone_verification_verified = None
+                user.phone_verification_token = None
+                return empty_pb2.Empty()
+
+            if not is_known_operator(phone):
+                context.abort(grpc.StatusCode.UNIMPLEMENTED, errors.UNRECOGNIZED_PHONE_NUMBER)
+
+            if user.phone_verification_sent and now() - user.phone_verification_sent < timedelta(days=180):
+                context.abort(grpc.StatusCode.RESOURCE_EXHAUSTED, errors.REVERIFICATION_TOO_EARLY)
+
+            token = sms.generate_random_code()
+            result = sms.send_sms(phone, sms.format_message(token))
+
+            if result == "success":
+                user.phone = phone
+                user.phone_verification_sent = now()
+                user.phone_verification_token = token
+                user.phone_verification_attempts = 0
+                return empty_pb2.Empty()
+
+        context.abort(grpc.StatusCode.UNIMPLEMENTED, result)
+
+    def VerifyPhone(self, request, context):
+        if len(request.token) != 6:
+            context.abort(grpc.StatusCode.INVALID_ARGUMENT, errors.WRONG_SMS_CODE)
+
+        with session_scope() as session:
+            user = session.query(User).filter(User.id == context.user_id).one()
+            if user.phone_verification_token is None:
+                context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.NO_PENDING_VERIFICATION)
+
+            if user.phone_verification_attempts > 3:
+                context.abort(grpc.StatusCode.RESOURCE_EXHAUSTED, errors.TOO_MANY_SMS_CODE_ATTEMPTS)
+
+            if user.phone_verification_token != request.token:
```

Comment:
This should be a secure compare, as insignificant as it is. (This is prone to timing attacks, though the 3 verification attempts mitigates that, still good to use a constant time compare). You can do it with `secure_compare`: that takes bytes though, you can either encode it here or add a helper function in crypto.py.

I would maybe just create `generate_sms_code()` and `verify_sms_code(request_token, right_token)` in `crypto.py`